test/jenkins.test.ts

```diff
@@ -0,0 +1,430 @@
+      seedJobs: seedJobs,
+    });
+
+    jenkins.addSeedJobs(...moreSeedJobs);
+
+    // THEN
+    const manifest = Testing.synth(chart);
+    expect(manifest).toMatchSnapshot();
+    expect(manifest[0].spec.seedJobs[0]).toEqual(seedJobs[0]);
+    expect(manifest[0].spec.seedJobs[1]).toEqual(moreSeedJobs[0]);
+  });
+
+  test('with all fields mentioned', () => {
+    // GIVEN
+    const app = Testing.app();
+    const chart = new Chart(app, 'test');
+
+    const namespace = 'jenkins-test-namespace';
+    const labels = { customApp: 'my-jenkins' };
+    const disableCSRFProtection = true;
+    const basePlugins = [{
+      name: 'configuration-as-code',
+      version: '1.55',
+    }];
+    const plugins = [{
+      name: 'simple-theme-plugin',
+      version: '0.7',
+    }];
+    // Seed Jobs Example: https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/deploying-jenkins/
+    const seedJobs = [{
+      id: 'jenkins-operator',
+      targets: 'cicd/jobs/*.jenkins',
+      description: 'Jenkins Operator repository',
+      repositoryBranch: 'master',
+      repositoryUrl: 'https://github.com/jenkinsci/kubernetes-operator.git',
+    }];
+
+    // WHEN
+    new Jenkins(chart, 'my-jenkins', {
+      metadata: {
+        namespace: namespace,
+        labels: labels,
+      },
+      disableCsrfProtection: disableCSRFProtection,
+      basePlugins: basePlugins,
+      plugins: plugins,
+      seedJobs: seedJobs,
+    });
+
+    // THEN
+    const manifest = Testing.synth(chart);
+
+    expect(manifest).toMatchSnapshot();
+    expect(manifest[0].metadata.namespace).toEqual(namespace);
+    expect(manifest[0].metadata.labels).toEqual(labels);
+    expect(manifest[0].spec.master.disableCSRFProtection).toEqual(disableCSRFProtection);
+
+    for (const plugin of manifest[0].spec.master.basePlugins) {
+      if (plugin.name == basePlugins[0].name) {
+        expect(plugin.version).toEqual(basePlugins[0].version);
+      }
+    }
+    expect(manifest[0].spec.master.plugins[0].name).toEqual(plugins[0].name);
+    expect(manifest[0].spec.master.plugins[0].version).toEqual(plugins[0].version);
+    expect(manifest[0].spec.seedJobs[0]).toEqual(seedJobs[0]);
+  });
+
+  test(('modify base plugin version using escape hatches'), () => {
+    // GIVEN
+    const app = Testing.app();
+    const chart = new Chart(app, 'test');
+    const basePlugins = [{
+      name: 'workflow-job',
+      version: '3.00',
+    }];
+
+    // WHEN
+    const jenkins = new Jenkins(chart, 'my-jenkins');
+    const jenkinsApiObject = ApiObject.of(jenkins);
+    jenkinsApiObject.addJsonPatch(JsonPatch.replace('/spec/master/basePlugins/1', basePlugins[0]));
+
+    // THEN
+    const manifest = Testing.synth(chart);
+
+    expect(manifest).toMatchSnapshot();
+
+    for (const plugin of manifest[0].spec.master.basePlugins) {
+      if (plugin.name == basePlugins[0].name) {
+        expect(plugin.version).toEqual(basePlugins[0].version);
+      }
+    }
+  });
+});
```

Comment:
Add a test for instantiating two instances with different configurations.

Reply:
Will add a test for this scenario.